daemon/src/test/scala/io/gearpump/cluster/scheduler/PrioritySchedulerSpec.scala

```diff
@@ -136,4 +135,42 @@ class PrioritySchedulerSpec(_system: ActorSystem) extends TestKit(_system) with
       mockAppMaster.expectMsg(5 seconds, ResourceAllocated(Array(ResourceAllocation(Resource(60), mockWorker1.ref, workerId1))))
     }
   }
+
+  "The PriorityScheduler" should {
+    "handle the resource request with different executor number" in {
```

Comment:
This is a long test, 36 lines. Please document some key steps so we know how the test works.